docs/basic-features/data-fetching/getServerSideProps.md

```diff
@@ -0,0 +1,112 @@
+---
+description: Fetch data on each request with `getServerSideProps`.
+---
+
+# `getServerSideProps`
+
+If you export an `async` function called `getServerSideProps` (server-side rendering) from a page, Next.js will pre-render this page on each request using the data returned by `getServerSideProps`.
+
+```js
+export async function getServerSideProps(context) {
+  return {
+    props: {}, // will be passed to the page component as props
+  }
+}
+```
+
+The [`getServerSideProps` API reference](/docs/api-reference/data-fetching/getServerSideProps.md) covers all parameters and props that can be used with `getServerSideProps`.
+
+## When should I use `getServerSideProps`?
+
+You should use `getServerSideProps` only if you need to pre-render a page whose data must be fetched at request time. [Time to First Byte (TTFB)](/learn/seo/web-performance) will be slower than [`getStaticProps`](/docs/basic-features/data-fetching/getStaticProps.md) because the server must compute the result on every request, and the result can only be cached by a CDN using `cache-control` headers (which could require extra configuration).
+
+If you do not need to pre-render the data, then you should consider fetching data on the [client side](#fetching-data-on-the-client-side).
+
+### `getServerSideProps` or API Routes?
+
+It can be tempting to reach for an [API Route](/docs/api-routes/introduction.md) when you want to fetch data from the server, then call that API route from `getServerSideProps`. This is an unnecessary and inefficient approach, as it will cause an extra request to be made due to both `getServerSideProps` and API Routes running on the server.
+
+Instead, directly import the logic used inside your API Route into `getServerSideProps`. This could mean calling a CMS, database, or other API directly from inside `getServerSideProps`.
+
+### TypeScript: Use `GetServerSideProps`
```

Comment:
these typescript sections feel slightly out of place but i can also understand the struggle to find the right spot for them. could you share your thinking here about the typescript api being more "concept" than "reference"? i am curious (i don't have a strong opinion about which it is)

Reply:
Oh wow, thats a mess up from me. They should be in the api ref docs. I have getServerSideProps, but didnt move the others. Thank you for spotting this!